File: src/main/java/com/wanted/preonboarding/ticket/domain/Reservation.java
package com.wanted.preonboarding.ticket.domain;

import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Getter
@Entity
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Table(name = "reservation")
public class Reservation {
    @Id
    @Column(name = "id")
    @EqualsAndHashCode.Include
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "performance_id")
    private Performance performance;
    
    @Column(name = "name")
    private String name;
    
    @Column(name = "phone_number")
    private String phoneNumber;
    
    @Column(name = "round")
    private int round;
    
    @Column(name = "gate")
    private int gate;
    
    @Column(name = "line")
    private String line;
    
    @Column(name = "seat")
    private int seat;

    @CreatedDate
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/domain/Reservation.kt
package com.wanted.preonboarding.ticket.domain

import java.time.LocalDateTime

class Reservation(
    val id: ReservationId = ReservationId(0),
    private val userInfo: UserInfo,
    private val seatInfo: SeatInfo,
    private val createdAt: LocalDateTime = LocalDateTime.now(),
    private val updatedAt: LocalDateTime? = null,
) {
    fun isSameSeat(seatInfo: SeatInfo): Boolean {
        return this.seatInfo == seatInfo
    }

    fun getUserInfo(): UserInfo {
        return userInfo
    }

    fun getSeatInfo(): SeatInfo {
        return seatInfo
    }

    fun isReserved(userInfo: UserInfo): Boolean {
        return this.userInfo == userInfo
    }
}
